Framework/Core/src/ConfigurationOptionsRetriever.cxx

```diff
@@ -34,7 +34,13 @@ ConfigurationOptionsRetriever::ConfigurationOptionsRetriever(std::vector<ConfigP
   : mCfg{cfg},
     mStore{}
 {
-  PropertyTreeHelpers::populate(schema, mStore, cfg->getRecursive(mainKey));
+  boost::property_tree::ptree in;
+  try {
+    in = cfg->getRecursive(mainKey);
+  } catch (...) {
+    in.clear();
+  }
+  PropertyTreeHelpers::populate(schema, mStore, in);
```

Comment:
@awegrzyn Is there a way to detect if a given node exists?

Reply:
I don't think so, this is limited by Consul backend as it anyway requires request in order to find out if node exists or not